tools/tests/on-disk-format-checker.sh

```diff
@@ -0,0 +1,164 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ------
+#
+# This script performs tests to ensure that no unintended breaking changes are
+# introduced to the on-disk format of the s3gw.
+# It works by spinning up an old version of the s3gw, uploading a couple objects
+# to it, then reusing that volume with a new version of the s3gw, trying to read
+# those objects again and ensuring integrity with checksums along the way.
+
+DEBUG=${DEBUG:-""}
+
+CEPH_DIR=${CEPH_DIR:-"$PWD/ceph"}
+OLD_VERSION=${OLD_VERSION:-"latest"}
+NEW_VERSION=${NEW_VERSION:-""}
+
+S3GW_HOST=${S3GW_HOST:-"127.0.0.1:7480"}
+
+VOL=$(mktemp -q -d s3gw.XXXX --tmpdir="$PWD")
+SRC=$(mktemp -q -d src.XXXX --tmpdir="$PWD")
+DST=$(mktemp -q -d dst.XXX --tmpdir="$PWD")
+CFG="$PWD/s3cfg"
+
+CONTAINER=
+
+
+_podman() {
+  if command -v podman ; then
+    podman "$@"
+  elif command -v docker ; then
+    docker "$@"
+  else
+    false
+    _cleanup
+  fi
+}
+
+debug() {
+  [ -z "$DEBUG" ] || echo "$@"
+}
+
+s3() {
+	s3cmd -q -c "$CFG" "$@"
+}
+
+_start_s3gw() {
+  local version="$1"
+
+  if [ -n "$version" ] ; then
+    debug "Running s3gw ${version}"
+    CONTAINER=$(_podman run \
+                  --rm -d \
+                  -v "${VOL}:/data" \
+                  -p 7480:7480 \
+                  "quay.io/s3gw/s3gw:${version}")
+  else
+    debug "Running s3gw from ${CEPH_DIR}/build/bin"
+    CONTAINER=$(_podman run \
+                  --rm -d \
+                  -v "${VOL}:/data" \
+                  -v "${CEPH_DIR}/build/bin:/radosgw/bin" \
+                  -v "${CEPH_DIR}/build/lib:/radosgw/lib" \
+                  -p 7480:7480 \
+                  quay.io/s3gw/run-radosgw \
+                    --rgw-backend-store sfs \
+                    --debug-rgw 1)
+  fi
+
+  for _ in {1..600} ; do
+    if curl -s "$S3GW_HOST" > /dev/null ; then
+      break
+    fi
+    sleep .1
+  done
+}
+
+_stop_s3gw() {
+  _podman kill "$CONTAINER"
+}
+
+_setup() {
+  debug "Setting up..."
+
+  cat > "$CFG" <<EOF
+[default]
+access_key = test
+secret_key = test
+host_base = ${S3GW_HOST}/
+host_bucket = ${S3GW_HOST}/%(bucket)
+signurl_use_https = False
+use_https = False
+signature_v2 = True
+signurl_use_https = False
+EOF
+
+  for i in {1..100} ; do
+    dd if=/dev/random bs=1k count=5k of="${SRC}/obj-${i}.bin" status=none
+  done
+
+  _start_s3gw "$OLD_VERSION"
+	s3 mb "s3://bucket"
+	s3 put "${SRC}"/* "s3://bucket/"
+  _stop_s3gw
+
+  _start_s3gw "$NEW_VERSION"
+}
+
+_sha256sums() {
+  local workdir="$1"
+  pushd "$workdir" > /dev/null || exit 1
+  sha256sum ./* > checksums
+  popd > /dev/null || exit 1
+}
+
+_cleanup() {
+  local result="$?"
+
+  _stop_s3gw
+  debug "Cleaning up..."
+	rm -rf "$VOL"
```

Comment:
Why is this indented?

Reply:
Accidental. :facepalm: I tabbed when I should have spaced and by pure coincidence my editor shows this with the same indentation in this case.